src/RuleContext.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Rbac;
+
+final class RuleContext
+{
+    public function __construct(private RuleFactoryInterface $ruleFactory, private array $parameters = [])
```

Comment:
```suggestion
    public function __construct(private RuleFactoryInterface $ruleFactory, private array $parameters)
```

Manager always pass parameters and it is only place where created context object.